src/Changelog/ChangelogReleaseNotes.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\AutomaticReleases\Changelog;
+
+use Phly\KeepAChangelog\Common\ChangelogEditor;
+use Phly\KeepAChangelog\Common\ChangelogEntry;
+use RuntimeException;
+use Webmozart\Assert\Assert;
+
+class ChangelogReleaseNotes
```

Comment:
I'd love for this to be `@psalm-immutable`, but we do perform filesystem interactions below.

It is still possible to achieve that by making the method that performs side effects `static`, but that isn't the end of the world, so I'll leave it up to the author's discretion

Reply:
So, I tried doing this just now, and the problem I ran into is that `ChangelogEntry` is mutable, and Psalm doesn't like having an externally mutable instance inside an immutable object. I tried cloning it, but it somehow fails to see that there's a clone created internally, and that the instance is not exposed publicly at any point (there's no getter/setter for it).

I think I'll move forward with it, but will need to suppress the `ImpurePropertyAssignment` warnings, knowing that they were being raised incorrectly. (Just letting you know in advance.)

Reply:
Ah, I see, `ChangelogEntry` is indeed external. Missed that, sorry!

Overall, I'm OK with suppressing those for now, so this version of the code should be fine.